packages/loot-core/src/server/filters/app.ts

```diff
@@ -56,50 +56,64 @@ async function filterNameExists(name, filterId, newItem) {
   return true;
 }
 
-//TODO: Possible to simplify this?
-//use filters and maps
 function conditionExists(item, filters, newItem) {
   const { conditions, conditionsOp } = item;
-  let condCheck = [];
-  let fCondCheck = false;
-  let fCondFound;
+  let fConditionFound = null;
 
-  filters.map(filter => {
+  filters.some(filter => {
     if (
-      !fCondCheck &&
-      //If conditions.length equals 1 then ignore conditionsOp
-      (conditions.length === 1 ? true : filter.conditionsOp === conditionsOp) &&
+      (conditions.length === 1 || filter.conditionsOp === conditionsOp) &&
       !filter.tombstone &&
       filter.conditions.length === conditions.length
     ) {
-      fCondCheck = false;
-      conditions.map((cond, i) => {
-        condCheck[i] =
-          filter.conditions.filter(fcond => {
-            return (
-              cond.value === fcond.value &&
-              cond.op === fcond.op &&
-              cond.field === fcond.field
-            );
-          }).length > 0;
-        fCondCheck = (i === 0 ? true : fCondCheck) && condCheck[i];
-        return true;
+      let allConditionsMatch = true;
+
+      conditions.some(cond => {
```

Comment:
```suggestion
      let allConditionsMatch = !conditions.some(cond => {
```

suggestion: eliminate the need to set the allConditionsMatch within this some function by just grabbing the result of `some`

Reply:
Thanks for the suggestion! Took it a step further and simplified that chunk of logic even more.